### ISSUE TITLE
Invalid URI exception on opening file in root directory

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#6104]
VS Mac: 7.7.0.1444
Git revision: 21c6b8485df5c695a53269ab999298dcba289e7c
Build date: 2018-09-26 15:59:10+00
Build branch: release-7.7
Xamarin extensions: 8c052293f9d925b55a5596022f1fd61e02601036

1. Open a project in VS Mac.
2. Open a file (not part of the project) that is in the root directory /

```
Error while getting coding conventions,
System.UriFormatException: Invalid URI: The format of the URI could not be determined.
  at System.Uri.CreateThis (System.String uri, System.Boolean dontEscape, System.UriKind uriKind) [0x0007d] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System/net/System/UriExt.cs:56 
  at System.Uri..ctor (System.String uriString) [0x00014] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/mcs/class/referencesource/System/net/System/URI.cs:408 
  at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache.NormalizePath (System.String path) [0x00000] in <33418e5da6fc40388dc1fe4aee49e148>:0 
  at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache+<GetConfigsAsync>d__14.MoveNext () [0x00036] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCache+<GetCodingConventionSnapshotAsync>d__8.MoveNext () [0x0009e] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.VisualStudio.CodingConventions.EditorConfig.EditorConfigCodingConventionsProvider+<GetCodingConventionsAsync>d__4.MoveNext () [0x00099] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.VisualStudio.CodingConventions.CodingConventionsManager+<GetConventionContextAsync>d__13.MoveNext () [0x0012b] in <33418e5da6fc40388dc1fe4aee49e148>:0 
--- End of stack trace from previous location where exception was thrown ---
  at MonoDevelop.Ide.Editor.EditorConfigService+<GetEditorConfigContext>d__4.MoveNext () [0x0005a] in /Users/vsts/agent/2.140.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/EditorConfigService.cs:51 
```

> VS bug [#692245](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/692245)

**Comment by Therzok** [comment]
Filed as https://devdiv.visualstudio.com/DevDiv/_queries/edit/657362/?triage=true

**Comment by mkrueger** [comment]
Seems to be an error in the class library

This crashes:

var uri = new Uri ("/a");

But not every file name.

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.655 (master)

Pull Request #6122 merged by: Lluis Sanchez
Author: mono
Commit: 369489a4a82cf8a94ff0b115f4d7a95a6d3e7b26 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1485 (release-7.7)

Pull Request #6125 merged by: Lluis Sanchez
Author: mono
Commit: dcd3f691181ba30d0ef2edf6c0924d2d04e6b4a3 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.197 (release-8.0)

Pull Request #6122 merged by: Lluis Sanchez
Author: mono
Commit: 369489a4a82cf8a94ff0b115f4d7a95a6d3e7b26 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6122 merged by: Lluis Sanchez
Author: mono
Commit: 369489a4a82cf8a94ff0b115f4d7a95a6d3e7b26 (xamarin/monodevelop)
